src/Tools/BuildValidator/CompilationDiff.cs

```diff
@@ -114,124 +130,183 @@ private CompilationDiff(ImmutableArray<Diagnostic> diagnostics, string originalP
+                    ILFilePath: Path.Combine(outputPath, assemblyName + ".il"),
+                    PdbXmlFilePath: Path.Combine(outputPath, assemblyName + ".pdb.xml"),
+                    CustomDataFilePath: Path.Combine(outputPath, "custom-data.txt"));
+
+                Directory.CreateDirectory(outputPath);
+                File.WriteAllBytes(assemblyFilePath, buildInfo.AssemblyBytes);
+
+                // This is deliberately named .extracted.pdb instead of .pdb. A number of tools will look
+                // for a PDB with the name assemblyName.pdb. Want to make explicitly sure that does not 
+                // happen and such tools always correctly fall back to the embedded PDB. 
+                var pdbFilePath = Path.Combine(outputPath, assemblyName + ".extracted.pdb");
+                writeAllBytes(pdbFilePath, new Span<byte>(buildInfo.PdbMetadataReader.MetadataPointer, buildInfo.PdbMetadataReader.MetadataLength));
+
+                createMetadataVisualization(buildDataFiles.AssemblyMdvFilePath, buildInfo.AssemblyMetadataReader);
+                createMetadataVisualization(buildDataFiles.PdbMdvFilePath, buildInfo.PdbMetadataReader);
+                createDataFile(buildDataFiles.CustomDataFilePath, buildInfo.AssemblyReader, buildInfo.PdbMetadataReader);
+
+                var pdbToXmlOptions = PdbToXmlOptions.ResolveTokens
+                    | PdbToXmlOptions.ThrowOnError
+                    | PdbToXmlOptions.ExcludeScopes
+                    | PdbToXmlOptions.IncludeSourceServerInformation
+                    | PdbToXmlOptions.IncludeEmbeddedSources
+                    | PdbToXmlOptions.IncludeTokens
+                    | PdbToXmlOptions.IncludeMethodSpans;
+
+                using var pdbXmlStream = File.Create(buildDataFiles.PdbXmlFilePath);
+                PdbToXmlConverter.ToXml(
+                    new StreamWriter(pdbXmlStream),
+                    pdbStream: new UnmanagedMemoryStream(buildInfo.PdbMetadataReader.MetadataPointer, buildInfo.PdbMetadataReader.MetadataLength),
+                    peStream: new MemoryStream(buildInfo.AssemblyBytes),
+                    options: pdbToXmlOptions,
+                    methodName: null);
+
+                Process.Start(@"C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\ildasm.exe", $@"{assemblyFilePath} /all /out={buildDataFiles.ILFilePath}").WaitForExit();
+
+                return buildDataFiles;
+            }
 
-                        PdbToXmlConverter.ToXml(
-                            new StreamWriter(originalPdbXml),
-                            pdbStream: new UnmanagedMemoryStream(optionsReader.PdbReader.MetadataPointer, optionsReader.PdbReader.MetadataLength),
-                            peStream: new MemoryStream(originalBytes),
-                            options: pdbToXmlOptions,
-                            methodName: null);
+            static unsafe void writeAllBytes(string filePath, Span<byte> span)
```

Comment:
>unsafe [](start = 19, length = 6)

Minor: Is `unsafe` needed?

Reply:
yep, fixed

Reply:
i mean "yep you're right, not needed, fixed"